DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/onchip_memory_cache/README.md

```diff
@@ -21,36 +21,36 @@ In DPC++ task kernels for FPGA, our objective is to achieve an initiation interv
 
 When the loop contains a loop-carried variable implemented in on-chip memory, the compiler often *cannot* achieve II=1 because the memory access takes more than one clock cycle. If the updated memory location may be needed on the next loop iteration, the next iteration must be delayed to allow time for the update, hence II > 1.
 
-The on-chip memory cache technique breaks this dependency by storing recently-accessed values in a cache capable of a 1-cycle read-modify-write operation. The cache is implemented in FPGA registers rather than on-chip memory. By pulling memory accesses preferentially from the register cache, the loop-carried dependency is broken.
+The technique of using the OnchipMemoryWithCache class breaks this dependency by storing recently-accessed values in a cache capable of a 1-cycle read-modify-write operation. The cache is implemented in FPGA registers rather than on-chip memory. By pulling memory accesses preferentially from the register cache, the loop-carried dependency is broken.
 
-### When is the on-chip memory cache technique applicable?
+### When is the on-chip memory with cache technique applicable?
 
 ***Failure to achieve II=1 because of a loop-carried memory dependency in on-chip memory***:
-The on-chip memory cache technique is applicable if the compiler could not pipeline a loop with II=1 because of an on-chip memory dependency. (If the compiler could not achieve II=1 because of a *global* memory dependency, this technique does not apply as the access latencies are too great.)
+The on-chip memory with cache technique is applicable if the compiler could not pipeline a loop with II=1 because of an on-chip memory dependency. (If the compiler could not achieve II=1 because of a *global* memory dependency, this technique does not apply as the access latencies are too great.)
```

Comment:
I'm not sure what such a diagram would look like, and I'm afraid I've spent more time on this already than I had planned, so that will have to wait for a future update (perhaps by you?)